pkg/lock/dir.go

```diff
@@ -143,18 +192,22 @@ func (l *DirLock) Fd() (int, error) {
 }
 
 // Close closes the lock which implicitly unlocks it as well
-func (l *DirLock) Close() error {
+func (l *FileLock) Close() error {
 	fd := l.fd
 	l.fd = -1
 	return syscall.Close(fd)
 }
 
-// NewLock opens a new lock on a directory without acquisition
-func NewLock(dir string) (*DirLock, error) {
-	l := &DirLock{dir: dir, fd: -1}
+// NewLock opens a new lock on a file without acquisition
+func NewLock(path string, isDir bool) (*FileLock, error) {
```

Comment:
Can you make NewLock private or change isDir from being a boolean to taking an enumeration.